pkg/controller/storage/account/claim.go

```diff
@@ -59,21 +52,16 @@ func (c *ClaimSchedulingController) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(claimscheduling.NewReconciler(mgr,
 			resource.ClaimKind(storagev1alpha1.BucketGroupVersionKind),
 			resource.ClassKind(v1alpha3.AccountClassGroupVersionKind),
+			claimscheduling.WithLogger(l.WithValues("controller", name)),
+			claimscheduling.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
 		))
 }
 
-// A ClaimDefaultingController reconciles Bucket claims that omit their resource
-// ref, class ref, and class selector by choosing a default Azure Account
-// resource class if one exists.
-type ClaimDefaultingController struct{}
-
-// SetupWithManager sets up the ClaimDefaultingController using the supplied
-// manager.
-func (c *ClaimDefaultingController) SetupWithManager(mgr ctrl.Manager) error {
-	name := strings.ToLower(fmt.Sprintf("defaulter.%s.%s.%s",
-		storagev1alpha1.BucketKind,
-		v1alpha3.AccountKind,
-		v1alpha3.Group))
+// SetupClaimDefaulting adds a controller that reconciles Bucket claims that
+// omit their resource ref, class ref, and class selector by choosing a default
+// Azure Account resource class if one exists.
+func SetupClaimDefaulting(mgr ctrl.Manager, l logging.Logger) error {
+	name := claimdefaulting.ControllerName(v1alpha3.AccountKind)
```

Comment:
container / account is the first place where we will run into conflicts with this naming pattern mentioned above. I think it is likely fine to have both claim controllers named as bucket claim controllers (since they are both reconciling them), what do you think?

Reply:
Yeah, I'm not too concerned here. As best as I can tell this controller name string is only used in logs, events, etc. I imagine we won't treat accounts as a kind of bucket once we solve our claims and composition challenges.